### DISCUSSION TITLE
Contributions for each variable per time period

### DISCUSSION THREAD

**Discussion by edwardtan2** [google/meridian#374] [Q&A]
I'm building a national geo model on a weekly time period and am looking to get the contributions (fitted values) per week for each of the explanatory variables - ie. controls, media channels. Is there a way to quickly retrieve this? This would be similar to the pymc marketing function 'mmm.compute_mean_contributions_over_time'. Any help would be greatly appreciated.

**Comment by brendalewisprice** [comment]
Meridian has a function in the analyzer class called [incremental_impact](https://developers.google.com/meridian/reference/api/meridian/analysis/analyzer/Analyzer#incremental_impact) that returns the contributions for each media channel for each draw. The default behavior is to return the impact values for the entire time period, so you would need to specify the time values of interest and take the mean or median of the draws to get the channel-specific estimates.

Below is a code example for accessing those values.

```
mmm_analyzer = analyzer.Analyzer(meridian)
mmm_analyzer.incremental_impact()
```

Because of lagged effects, we don't recommend outputting the incremental impact for individual media at the weekly level. If you do repeated aggregations over time, you may want to make sure the aggregations are at least `max_lag` number of weeks long.  Additionally, because none of the media parameters vary over time, the value for doing this is limited and may only be best used as a model diagnostic tool.

As for control variables, they are mainly used to determine the baseline, but they are not attributed with a contribution to the KPI. If you want to estimate a causal effect for a variable, you can consider including it as a [non-media treatment](https://developers.google.com/meridian/docs/advanced-modeling/organic-and-non-media-variables?hl=en#deciding_if_a_variable_is_a_non-media_treatment_variable_or_a_control).

**Comment by andyl7an** [comment]
Hey @edwardtan2 !

For visualizing weekly contributions, check out the [`visualizer.MediaSummary.plot_channel_contribution_area_chart`](https://developers.google.com/meridian/reference/api/meridian/analysis/visualizer/MediaSummary#plot_channel_contribution_area_chart) function we added in `v1.0.8`.

It generates a stacked area chart showing the estimated contribution of the baseline and each channel for every week. This provides that decompositional view over time you're looking for, showing how the modelled total breaks down into its core drivers.

You can see an example here: [meridian-getting-started](https://developers.google.com/meridian/notebook/meridian-getting-started#step_4_generate_model_results_two-page_output)
